src/fields.jl

```diff
@@ -108,15 +65,26 @@ struct EdgeField{A<:AbstractArray, G<:Grid} <: Field
     grid::G
 end
 
-function EdgeField(::CPU, g::RegularCartesianGrid{T, <:AbstractRange}) where T <: AbstractFloat
-    data = zeros(T, size(g))
-    EdgeField{typeof(data), typeof(g)}(data, g)
+EdgeField(T, arch, grid) = EdgeField(zeros(T, arch, grid), grid)
+EdgeField(arch, grid) = EdgeField(zeros(arch, grid), grid)
+
+abstract type Location end
+struct Cell <: Location end
+struct Interface <: Location end
+
+"""
```

Comment:
@jm-c See the new `GeneralField` type here.

Reply:
Note that once this is done we should also be able to eliminate `CellField` and `FaceFieldX`, etc, replacing them with `GeneralField` (and possibly renaming to just "`Field`"). 

This should be easy via the definition of 

`const CellField = GeneralField{C, C, C}`, 

with no other code changes, I think. Which is nice.